### DISCUSSION TITLE
[Chapter 07] AOP에 적용된 패턴은?

### DISCUSSION THREAD

**Discussion by ahngilwoong** [cowaine/spring-boot-for-msa#78] [Chapter 07. Spring AOP, Testing, and Auto Confiugration]
AOP는 결론적으로 관심사를 구분하여
어떠한 로직 전 후로 공통된 작업을 합니다
여기서 디자인 패턴들이 떠오르는데요.
대표적으로 프록시 패턴과 데코레이터 패턴이
떠오릅니다. 여기서 AOP에 적용된 패턴은 무슨 패턴일까요?

**Comment by JoisFe** [comment]
https://docs.spring.io/spring-framework/reference/core/aop/introduction-proxies.html

**Comment by Irisation23** [comment]
### 금주 CheckPoint

- AOP에 적용된 디자인 패턴에 대한 정리 필요

**Comment by ahngilwoong** [comment]
AOP는 내부에 실제 객체를 가지고 있지 않은 대신 실제 객체를 참조하는 참조 변수를 가지고 있다고 하는데 프록시 객체의 메서드를 호출하고 나면, 프록시 객체는 실제 객체를 참조하는 참조 변수를 통해 실제 객체의 메서드를 호출한다는 점에서 AOP와 같기 때문에 **프록시** 패턴이 맞다고 합니다.

AOP 프록시 패턴은 공통된 기능을 일괄적으로 추가하거나 수정하는 방식으로 동작하며, 데코레이터 패턴은 개별적으로 기능을 추가하거나 수정하는 방식으로 동작한다고 합니다.

참조 https://mumomu.tistory.com/175
